src/sentry/utils/committers.py

```diff
@@ -254,11 +246,22 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
 
     for committer in committers:
         commit_ids = [commit.id for (commit, _) in committer["commits"]]
-        committer["commits"] = [serialized_commits_by_id[commit_id] for commit_id in commit_ids]
+        commits_result = [serialized_commits_by_id[commit_id] for commit_id in commit_ids]
+        # Deduplicate commits
+        committer["commits"] = dedupeCommits(commits_result)
 
     metrics.incr(
         "feature.owners.has-committers",
         instance="hit" if committers else "miss",
         skip_internal=False,
     )
     return committers
+
+
+def dedupeCommits(commits):
+    result = {}
+    for obj in commits:
+        if obj["id"] not in result:
```

Comment:
@NisanthanNanthakumar Nit: you could have just done:
```
result = {}
for obj in commits:
  result[obj["id"]] = obj
```

Reply:
This isn't really a nit, it's an objective improvement in readability. Going even further:

```
    return {
        c["id"]: c
        for c in commits
    }.values()
```